scheduler/src/cook/mesos.clj

```diff
@@ -75,9 +78,92 @@
 
 (counters/defcounter [cook-mesos mesos mesos-leader])
 
+(defn make-mesos-driver
+  "Creates a mesos driver
+
+   Parameters:
+   mesos-framework-name     -- string, name to use when connecting to mesos.
+                               Will be appended with version of cook
+   mesos-role               -- string, (optional) The role to connect to mesos with
+   mesos-principal          -- string, (optional) principal to connect to mesos with
+   gpu-enabled?             -- boolean, (optional) whether cook will schedule gpu jobs
+   mesos-failover-timeout   -- long, (optional) time in milliseconds mesos will wait
+                               for framework to reconnect.
+                               See http://mesos.apache.org/documentation/latest/high-availability-framework-guide/
+                               and search for failover_timeout
+   mesos-master             -- str, (optional) connection string for mesos masters
+   scheduler                -- mesos scheduler implementation
+   framework-id             -- str, (optional) Id of framework if it has connected to mesos before
+   "
+  ([config scheduler]
+   (make-mesos-driver config scheduler nil))
+  ([{:keys [mesos-framework-name mesos-role mesos-principal gpu-enabled?
+            mesos-failover-timeout mesos-master] :as config}
+    scheduler framework-id]
+   (apply mesomatic.scheduler/scheduler-driver
+          scheduler
+          (merge
+            {:user ""
+             :name (str mesos-framework-name
+                        "-"
+                        cook.util/version)
+             :checkpoint true}
+            (when mesos-role
+              {:role mesos-role})
+            (when mesos-principal
+              {:principal mesos-principal})
+            (when gpu-enabled?
+              {:capabilities [{:type :framework-capability-gpu-resources}]})
+            (when mesos-failover-timeout
+              {:failover-timeout mesos-failover-timeout})
+            (when framework-id
+              {:id {:value framework-id}}))
+          mesos-master
+          (when mesos-principal
+            [{:principal mesos-principal}]))))
+
+(defn get-mesos-utilization
+  "Queries the mesos master to get the utilization.
+   Returns the max of cpu and mem utilization as a decimal (e.g. 0.92)"
+  [mesos-master-hosts]
+  (let [mesos-master-urls (map #(str "http://" % ":5050/metrics/snapshot") mesos-master-hosts)
```

Comment:
Sorry, thought I responded to this last time. Yes, but I don't think we should in this PR. This function was moved from another namespace (cook.mesos.rebalancer) because it was nicer from an injection standpoint to have it in cook.mesos. However, changing the functionality of it is unrelated to this pr.